iceoryx_dust/test/moduletests/test_container_fixed_position_container.cpp

```diff
@@ -92,6 +94,326 @@ TEST_F(FixedPositionContainer_test, Capacity)
     EXPECT_THAT(sut.capacity(), Eq(CAPACITY));
 }
 
+// BEGIN test copy constructor
+
+TEST_F(FixedPositionContainer_test, UsingCopyCtorEmptyContainerResultsInEmptyContainer)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "6c528ef3-9c2d-4eb2-93a9-2d998d0db380");
+
+    SutComplex copy_sut_complex{sut_complex};
+
+    EXPECT_THAT(copy_sut_complex.empty(), Eq(true));
+}
+
+TEST_F(FixedPositionContainer_test, UsingCopyCtorSingleElementContainerPreservesElement)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "f3aaf452-77fa-4535-bf0b-37bedefc2bf6");
+
+    constexpr DataType EXPECTED_VALUE{42U};
+    constexpr SutComplex::IndexType EXPECTED_SIZE{1U};
+
+    sut_complex.emplace(EXPECTED_VALUE);
+    SutComplex copy_sut_complex{sut_complex};
+
+    EXPECT_THAT(copy_sut_complex.size(), Eq(EXPECTED_SIZE));
+
+    // actually call copyAssignment
+    EXPECT_THAT(copy_sut_complex.begin()->stats.copyAssignment, Eq(EXPECTED_SIZE));
```

Comment:
You can use just `stats`. It's a static member and all elements share the same value.
```suggestion
    EXPECT_THAT(stats.copyAssignment, Eq(EXPECTED_SIZE));
```

Please also change further down.